src/main/java/io/cdap/plugin/snowflake/sink/batch/StructuredRecordToCSVRecordTransformer.java

```diff
@@ -111,12 +114,31 @@ public static String convertSchemaFieldToString(Object value, Schema.Field field
     switch (fieldSchema.getType()) {
       // convert to json so it can be saved to Snowflake's variant
       case RECORD:
-        return StructuredRecordStringConverter.toJsonString((StructuredRecord) value);
-      // convert to json so it can be saved to Snowflake's variant
+        try {
+          return StructuredRecordStringConverter.toJsonString((StructuredRecord) value);
+        } catch (IOException e) {
+          String errorMessage = String.format(
+            "Failed to encode record to JSON for schema field '%s' of type RECORD. Cause: %s",
+            field.getName(), e.getMessage()
+          );
+          throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+            errorMessage, String.format("Error message: %s", errorMessage), ErrorType.SYSTEM, true, e);
```

Comment:
please don't do redundant `ErrorMessage: %s`, this comment applies to whole PR.

Reply:
Here also, `errorReason` should be short and enough descriptive. `errorMessage` can contain more detail.

```
errorReason = Failed to encode record to JSON for schema field '%s' of type RECORD

errorMessage = Failed to encode record to JSON for schema field '%s' of type RECORD with message: 
```

Reply:
This comment applies to whole PR

Reply:
updated